ui/components/password-reset-form.js

```diff
@@ -0,0 +1,77 @@
+import { useRouter } from 'next/router'
+import { useState } from 'react'
+import { useSWRConfig } from 'swr'
+
+export default function PasswordResetForm() {
+  const { mutate } = useSWRConfig()
+  const router = useRouter()
+  const { token } = router.query
+
+  const [password, setPassword] = useState('')
+  const [error, setError] = useState('')
+
+  async function onSubmitSetPassword(e) {
+    e.preventDefault()
+
+    try {
+      const res = await fetch('/api/password-reset', {
+        method: 'post',
+        body: JSON.stringify({
+          token,
+          password,
+        }),
+      })
+
+      if (!res.ok) {
+        throw await res.json()
+      }
+
+      await res.json()
+
+      await mutate('/api/users/self')
+      router.replace('/')
+    } catch (e) {
+      setError(e)
```

Comment:
are we sure the `e` is always a string?

Reply:
no idea! This code was copied from ui/pages/login/index.js

Reply:
`setError(e.message)`